kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/model/VirtualClusterTest.java

```diff
@@ -42,6 +42,7 @@ void shouldBuildSslContext() {
                 false);
 
         // Then
+        assertThat(virtualCluster).isNotNull().extracting("downstreamSslContext").isNotNull();
```

Comment:
I appreciate you are following the existing pattern here but I've been preferring `satisfies` over extracting recently. So I started trying to write a fuller test locally using that. 
```suggestion
        assertThat(virtualCluster)
                .isNotNull()
                .satisfies(vc ->
                        assertThat(vc.getDownstreamSslContext())
                                .isPresent()
                                .hasValueSatisfying(
                                        sslContext -> {
                                            assertThat(sslContext.isClient()).isFalse();
                                            assertThat(sslContext.isServer()).isTrue();
                                            final SSLEngine sslEngine = sslContext.newEngine(ByteBufAllocator.DEFAULT);
                                            assertThat(sslEngine.getWantClientAuth()).isTrue(); //Depending on ClientAuth config
                                            assertThat(sslEngine.getNeedClientAuth()).isTrue(); //Depending on ClientAuth config
                                        }))
                .extracting("upstreamSslContext").isNotNull();

```

Which lead me to a couple of realisations. 

1. the `isNotNull` assertion isn't very helpful we should be checking the Optional isPresent instead (thats an existing bug)
2. This stuff is hard to test should but given the variability of `want` or `need` client auth I think it deserves its own test.

Reply:
Yeah, I'd tried to keep it light touch originally as new to the code, but can take another look at the tests.

Main issue I had yesterday was expecting the build to generate `withClientAuth` and `withNewClientAuth` methods on the config builders off the back of `withNewTls` builder method based on the new `clientAuth` String I'd added into Tls.java so I could do the integration tests, but it wasn't being generated. I was trying to work out if I'd missed something there, but I'll wait until you have a think on how to map our configuration with the `ClientAuth` netty Enum